templates/cluster.yml.j2

```diff
@@ -50,7 +51,7 @@ spec:
     - {{ ip }}
 {% endfor %}
   kubernetesVersion: {{ cluster.version | default(kops_default_version) }}
-  masterPublicName: api.{{ cluster.name }}
+  masterPublicName: {{ cluster.masterPublicName }}
```

Comment:
This is a drastic change and must be documented in the README. This will also trigger a major version upgrade as it will change the behaviour of current implementation.

So either make this a mandatory key and reflect that in the readme or make this an optional key and fault back to the previous implementation, but also reflect that in the readme.